lib/spack/spack/cmd/help.py

```diff
@@ -20,6 +20,7 @@
   package [constraints] [^dependency [constraints] ...]
 
   package                           any package from 'spack list'
```

Comment:
If you're going to put `/hash` on its own line, the first line should probably say:

```
        [package | /hash] [constraints] [^dependency [constraints] ...]
```

But I think that makes it harder to read, and it would be even harder if you did the same for `dependency`.  How about changing it to read like this:

```
    package                           any package name from 'spack list', or
    /hash                             unique prefix or full hash of installed package
```

Then do the same for dependencies:
```
^dependency [constraints]     specify constraints on dependencies
^/hash                        build with a specific installed dependency
```

And make `/hash` gray like it appears in `spack find -l`.